pennylane/templates/subroutines/qsvt.py

```diff
@@ -464,3 +466,290 @@ def _qsp_to_qsvt(angles):
+
+    **Example**
+
+    This example applies the polynomial :math:`P(x) = x - \frac{x^3}{2} + \frac{x^5}{3}` to a block-encoding
+    of :math:`x = 0.2`.
+
+    .. code-block::
+
+        poly = [0, 1.0, 0, -1/2, 0, 1/3]
+
+        qsp_angles = qml.poly_to_angles(poly, "QSP")
+        qsvt_angles = qml.transform_angles(qsp_angles, "QSP", "QSVT")
+
+        x = 0.2
+
+        # Encodes x in the top left of the matrix
+        block_encoding = qml.RX(2 * np.arccos(x), wires=0)
+
+        projectors = [qml.PCPhase(angle, dim=1, wires=0) for angle in qsvt_angles]
+
+        @qml.qnode(qml.device("default.qubit"))
+        def circuit_qsvt():
+            qml.QSVT(block_encoding, projectors)
+            return qml.state()
+
+        output = qml.matrix(circuit_qsvt, wire_order=[0])()[0, 0]
+        expected = sum(coef * (x**i) for i, coef in enumerate(poly))
+
+        print("output qsvt: ", output.real)
+        print("P(x) =       ", expected)
+
+    .. code-block:: pycon
+
+        output qsvt:  0.19610666666647059
+        P(x) =        0.19610666666666668
+    """
+
+    if routine1 == "QSP" and routine2 == "QSVT":
+        num_angles = len(angles)
+        update_vals = np.empty(num_angles)
+
+        update_vals[0] = 3 * np.pi / 4 - (3 + len(angles) % 4) * np.pi / 2
+        update_vals[1:-1] = np.pi / 2
+        update_vals[-1] = -np.pi / 4
+        update_vals = qml.math.convert_like(update_vals, angles)
+
+        return angles + update_vals
+
+    if routine1 == "QSVT" and routine2 == "QSP":
+        num_angles = len(angles)
+        update_vals = np.empty(num_angles)
+
+        update_vals[0] = 3 * np.pi / 4 - (3 + len(angles) % 4) * np.pi / 2
+        update_vals[1:-1] = np.pi / 2
+        update_vals[-1] = -np.pi / 4
+        update_vals = qml.math.convert_like(update_vals, angles)
+
+        return angles - update_vals
+
+    raise AssertionError(
+        f"Invalid conversion. The conversion between {routine1} --> {routine2} is not defined."
+    )
+
+
+def poly_to_angles(poly, routine, angle_solver="root-finding"):
+    r"""
+    Converts a given polynomial into angles for to be used with specific quantum signal processing (QSP)
+    or quantum singular value transformation (QSVT) routines.
+
+    Args:
+        poly (array-like): coefficients of the polynomial, ordered from lowest to higher degree.
+                           The polynomial must have defined parity and real coefficients.
+                           It also must meet that :math:`|P(x)| \leq 1` for all :math:`x \in [-1, 1]`.
+                           More information about these restriction can be
+                           found in [`arXiv:2105.02859 <https://arxiv.org/abs/2105.02859>`_]
+
+        routine (str): specifies the type of angle transformation required. Must be either: ``"QSP"`` or ``"QSVT"``
+
+        angle_solver (str): optional string which specifies the method used to calculate the angles (must be ``"root-finding"``)
```

Comment:
Please revise the descriptions. e.g., "Default is ..." not "must be ..." in parentheses.